eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -430,10 +429,10 @@ case class Commitment(fundingTxIndex: Long,
     // we allowed mismatches between our feerates and our remote's as long as commitments didn't contain any HTLC at risk
     // we need to verify that we're not disagreeing on feerates anymore before offering new HTLCs
     // NB: there may be a pending update_fee that hasn't been applied yet that needs to be taken into account
-    val localFeeratePerKw = feeConf.getCommitmentFeerate(feerates, params.remoteNodeId, params.commitmentFormat, capacity)
-    val remoteFeeratePerKw = localCommit.spec.commitTxFeerate +: changes.remoteChanges.all.collect { case f: UpdateFee => f.feeratePerKw }
-    remoteFeeratePerKw.find(feerate => feeConf.feerateToleranceFor(params.remoteNodeId).isFeeDiffTooHigh(params.commitmentFormat, localFeeratePerKw, feerate)) match {
-      case Some(feerate) => return Left(FeerateTooDifferent(params.channelId, localFeeratePerKw = localFeeratePerKw, remoteFeeratePerKw = feerate))
+    val localFeerate = feeConf.getCommitmentFeerate(feerates, params.remoteNodeId, params.commitmentFormat, capacity)
+    val remoteFeerate = localCommit.spec.commitTxFeerate +: changes.remoteChanges.all.collect { case f: UpdateFee => f.feeratePerKw }
+    remoteFeerate.find(feerate => feeConf.feerateToleranceFor(params.remoteNodeId).isProposedFeerateTooLow(params.commitmentFormat, localFeerate, feerate)) match {
```

Comment:
Why do you call directly `isProposedFeerateTooLow` here, and not `isFeeDiffTooHigh`?

Reply:
Because this isn't an issue if the feerate is too high at that point. What we want to avoid is having an HTLC in a commitment transaction that has a very low feerate, which we won't be able to confirm in time to claim the HTLC, so we only need to check that the feerate isn't too low.

Reply:
I found this surprising too and it took me some time to see why, maybe a comment in the code to explain why the check here  is different from the one we do when we check `update_fee` could help ? otherwise LGTM

Reply:
```suggestion
    // What we want to avoid is having an HTLC in a commitment transaction that has a very low feerate, which we won't be able to confirm in time to claim the HTLC, so we only need to check that the feerate isn't too low
    remoteFeerate.find(feerate => feeConf.feerateToleranceFor(params.remoteNodeId).isProposedFeerateTooLow(params.commitmentFormat, localFeerate, feerate)) match {
```

Reply:
Added comment in https://github.com/ACINQ/eclair/pull/2815/commits/5c9a5e881d771da483bf58beb96307943deee32d